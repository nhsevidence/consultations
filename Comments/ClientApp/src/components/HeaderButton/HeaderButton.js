// @flow

import React, {Fragment, PureComponent} from "react";
import {Link} from "react-router-dom";

type PropsType = {
	isAuthorised: boolean,
	responseCount?: number,
	consultationId: number,
}

export class HeaderButton extends PureComponent<PropsType> {

	render() {

		const showButton = ((this.props.isAuthorised) && 
							(typeof(this.props.responseCount) != undefined && this.props.responseCount != null && this.props.responseCount > 0));

		return (
			<Fragment>
				{showButton ? 
					<Link to={`/${this.props.consultationId}/review`}>Hello</Link>					
					: null}
			</Fragment>
		);
	}
}