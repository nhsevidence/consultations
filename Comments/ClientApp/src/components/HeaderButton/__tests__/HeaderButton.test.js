import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { HeaderButton }  from "../HeaderButton";


describe("[ClientApp] ", () => {
	describe("HeaderButton ", () => {
		it("link is found when user is authenticated and there are responses", () => {
		
			const wrapper = shallow(
				<HeaderButton isAuthorised={true} responseCount={1} />
			);
		  	expect(wrapper.find("Link").length).toEqual(1);
		});

		it("link is not found when user is not authenticated and there are responses", () => {
		
			const wrapper = shallow(
				<HeaderButton isAuthorised={false} responseCount={1} />
			);
		  	expect(wrapper.find("Link").length).toEqual(0);
		});

		it("link is not found when user is authenticated and the response count is undefined", () => {
		
			const wrapper = shallow(
				<HeaderButton isAuthorised={false} responseCount={undefined} />
			);
		  	expect(wrapper.find("Link").length).toEqual(0);
		});

		it("link is not found when user is authenticated and the response count is null", () => {
		
			const wrapper = shallow(
				<HeaderButton isAuthorised={false} responseCount={null} />
			);
		  	expect(wrapper.find("Link").length).toEqual(0);
		});

		it("link is found and has correct url to the review page", () => {
		
			const wrapper = shallow(
				<HeaderButton isAuthorised={true} responseCount={1} consultationId={22} />
			);
		  	expect(wrapper.find("Link").props().to).toEqual("/22/review");
		});

		it("should match snapshot", () => {
		
			const wrapper = shallow(
				<HeaderButton isAuthorised={true} responseCount={1} consultationId={22} />
			);
			expect(
				toJson(wrapper, {
					noKey: true,
					mode: "deep",
				})
			).toMatchSnapshot();
		});
	});
});
