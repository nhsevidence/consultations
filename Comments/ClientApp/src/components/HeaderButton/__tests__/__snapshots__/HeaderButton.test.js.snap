// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ClientApp]  HeaderButton  should match snapshot 1`] = `
<Fragment>
  <Link
    replace={false}
    to="/22/review"
  >
    Hello
  </Link>
</Fragment>
`;
